itest/mint_fund_seal_test.go

```diff
@@ -0,0 +1,706 @@
+		},
+	}
+
+	// Now, let's build our asset requests. We'll modify each of the
+	// standard asset requests used across the itests.
+	// Asset 0 will have no asset group, and an external script key with a
+	// tapscript root.
+	assetReqWithScriptKey := CopyRequest(simpleAssets[0])
+	assetReqWithScriptKey.Asset.ScriptKey = taprpc.MarshalScriptKey(
+		tweakedScript,
+	)
+
+	// Asset 1 will have a specific internal group key with a tapscript
+	// root. This asset will be a group anchor.
+	assetReqGroupedInternalTweaked := CopyRequest(simpleAssets[1])
+	assetReqGroupedInternalTweaked.Asset.NewGroupedAsset = true
+	assetReqGroupedInternalTweaked.Asset.GroupInternalKey = taprpc.
+		MarshalKeyDescriptor(managedGroupInternal)
+	assetReqGroupedInternalTweaked.Asset.
+		GroupTapscriptRoot = groupInternalTweak[:]
+
+	// Asset 2 will have an external group key with a tapscript root.
+	assetReqGroupedExternal := CopyRequest(issuableAssets[0])
+	assetReqGroupedExternal.Asset.GroupInternalKey = taprpc.
+		MarshalKeyDescriptor(groupExternalDesc)
+	assetReqGroupedExternal.Asset.GroupTapscriptRoot = groupExternalTweak[:]
+
+	// Asset 3 will be a member of the asset group created by Asset 1.
+	assetReqGroupMember := CopyRequest(issuableAssets[1])
+	assetReqGroupMember.Asset.NewGroupedAsset = false
+	assetReqGroupMember.Asset.GroupedAsset = true
+	assetReqGroupMember.Asset.GroupAnchor = assetReqGroupedInternalTweaked.
+		Asset.Name
+
+	assetReqs := []*mintrpc.MintAssetRequest{
+		assetReqWithScriptKey, assetReqGroupedInternalTweaked,
+		assetReqGroupedExternal, assetReqGroupMember,
+	}
+
+	// Let's fund a batch, adding the multisig tapscript sibling we
+	// constructed earlier.
+	fundReq := &mintrpc.FundBatchRequest{
+		BatchSibling: &siblingReq,
+	}
+	fundResp, err := aliceTapd.FundBatch(ctxt, fundReq)
+	require.NoError(t.t, err)
+	require.NotEmpty(t.t, fundResp.Batch)
+	require.Equal(
+		t.t, mintrpc.BatchState_BATCH_STATE_PENDING,
+		fundResp.Batch.State,
+	)
+	require.NotEmpty(t.t, fundResp.Batch.BatchPsbt)
+
+	// Now we can add all the asset requests created above.
+	BuildMintingBatch(t.t, aliceTapd, assetReqs)
+
+	// If we request a verbose list of batches, we should receive asset
+	// group information for exactly 3 assets.
+	listBatchResp, err := aliceTapd.ListBatches(
+		ctxt, &mintrpc.ListBatchRequest{
+			Verbose: true,
+		})
+	require.NoError(t.t, err)
+	require.Len(t.t, listBatchResp.Batches, 1)
+
+	unsealedAssets := listBatchResp.Batches[0].UnsealedAssets
+	unsealedAssetCount := fn.Count(
+		unsealedAssets, func(a *mintrpc.UnsealedAsset) bool {
+			return a.GroupKeyRequest != nil &&
+				a.GroupVirtualTx != nil
+		},
+	)
+	require.Equal(t.t, 3, unsealedAssetCount)
+
+	// Fetch the responses that match the three grouped asset requests.
+	assetGroupedInternalTweaked, err := fn.First(
+		unsealedAssets, func(a *mintrpc.UnsealedAsset) bool {
+			return a.Asset.Name ==
+				assetReqGroupedInternalTweaked.Asset.Name
+		})
```

Comment:
nit: closing parenthesis on next line (here and a couple of instances further below).